_posts/2016-09-23-Monitoring-with-Prometheus.md

```diff
@@ -0,0 +1,667 @@
+---
+layout: post
+authors: [tom_verelst]
+title: 'Monitoring with Prometheus'
+image: /img/prometheus.jpg
+tags: [Prometheus]
+category: Monitoring
+comments: true
+---
+
+It is needless to say the world is shifting towards DevOps and microservices.
+This holy grail we all aim for adds a lot of complexity,
+as well for monitoring.
+Instead of having to monitor one system,
+we are suddenly faced with the challenge to oversee our manifold services.
+There are many monitoring systems available,
+but not all of them are fit for monitoring large, distributed systems.
+
+Black box monitoring systems like [Nagios](https://www.nagios.org) allow you to check if an application is alive and healthy.
+This is done by e.g. pinging the service,
+checking if there is enough disk space,
+or monitoring the CPU usage.
+In a world of distributed architectures where high availability and fast response times are key,
+it is not sufficient to be only aware if a service is alive.
+It is crucial to know how a service is working internally as well.
+How many HTTP requests is it receiving?
+Are they handled correctly?
+How fast are requests handled for different endpoints?
+Are there many errors being logged?
+How many disk IO operations is the service making?
+These are all important questions that need to be asked to keep a service functional.
```

Comment:
that need to be asked => that need to be monitored in order to keep a service functional.